### DISCUSSION TITLE
[Tips] For ReadWise user and Thino Pro users | 从 ReadWise 笔记中获取内容

### DISCUSSION THREAD

**Discussion by Quorafind** [Quorafind/Obsidian-Thino#536] [Show and tell 展示方案]
For users of Thino Pro and Readwise, Thino 2.4.23 and later support indexing of single-file content with custom tags, so when you use the Readwise plugin and set the template to the following mode, Thino can directly index all Readwise saved content:

对于 Thino Pro 以及 Readwise 用户，在 Thino 2.4.23 以后，支持索引自定义标签的单文件内容，因此，当你使用 Readwise 插件，且设置模板为以下的模式时，Thino 可以直接索引其中的所有 Readwise 保存内容：

You can set the default export template of Readwise here: https://readwise.io/export/obsidian/preferences

你可以设置 Readwise 的默认导出模板：https://readwise.io/export/obsidian/preferences

You can set the template as follows:

设置模板时依次的每个空(对应位置)建议填写：

Set `Page metadata` as follows

设置页面元数据模板如下

```markdown
---
title: {{full_title}}
author: {{author}}
bookId: "{{book_id}}"
{% if image_url -%}
cover: "![rw-book-cover]({{image_url}})"
{% endif -%}
{% if url -%}
url: {{url}}
{% endif -%}
{% if source -%}
source: {{source}}
{% endif -%}
tags: ["thino/file"]
readingDate: {{date}}
lastReadDate: {{last_highlighted_date|date('Y-m-d')}}
---

{% if image_url -%}
![rw-book-cover]({{image_url}})

{% endif -%}
Author:: {% if author %}[[{{author}}]]{% endif %}
{% if document_tags -%}
Document Tags:: {% for tag in document_tags %}#{{tag}} {% endfor %}
{% endif -%}
{% if category -%}
Category:: [[{{category}}]]
{% endif -%}
```

Set `Highlight` as follows(Notice the blank line at the bottom)

设置高亮数据模板如下（注意底部的空行）

```markdown
> [!thino] {% if highlight_date %}{{highlight_date|date('Y/m/d H:i:s')}} {% else -%}2024/04/01 00:00:00 {% endif -%} %% [id::{{ highlight_id }}] [thinoType::JOURNAL] %%
> 📌 {{ highlight_text }}{% if highlight_location and highlight_location_url %} ([{{highlight_location}}]({{highlight_location_url}})){% elif highlight_location %} ({{highlight_location}}){% endif %}{% if highlight_tags %} {% for tag in highlight_tags %} #{{tag}}{% endfor %}{% endif %} #ReadWise {% if highlight_note %}
> - 💭 Note:{{ highlight_note }}{% endif %}

```

Other keep default.

其他保持默认即可

It works as follows:

其中效果如下：

![Clip_2024-04-15_16-12-07](https://github.com/Quorafind/Obsidian-Thino/assets/13215013/7a1c3ec2-ecbd-40c0-bd2d-26335a2f8817)

**Comment by Drmiaomiao** [comment]
谢谢bon,昨晚刚在想自己写个模板，今天就把饭喂到嘴里了，page title部分需要把默认的内容清空，不然会出现yaml区识别出错。要是bonbon有空再写个五彩的模板就更好了！
![image](https://github.com/Quorafind/Obsidian-Thino/assets/87026440/404d3173-ac25-4df1-b750-ede138da26d6)
